src/common/Util.h

```diff
@@ -181,6 +181,29 @@ template<typename T> class uninitialized {
 	uninitialized& operator=(uninitialized&&) = delete;
 };
 
+// Use to perform an action only if a minimum time has passed since the last time it was performed.
+// Not thread-safe.
+class MinimumDelay {
+public:
+	// duration: length of delay in milliseconds
+	MinimumDelay(int duration): duration(duration), lastTime(std::numeric_limits<int>::min()) {}
+
+	// Returns true if it is OK to perform the action again.
+	bool check() {
```

Comment:
const function. Also, I think our style has us capitalizing function names.

Reply:
It is a key part of this method's contract that the internal state is modified.